app/src/main/java/com/github/polypoly/app/network/storable/StorableObject.kt

```diff
@@ -0,0 +1,104 @@
+package com.github.polypoly.app.network.storable
+
+import kotlin.reflect.KClass
+
+/**
+ * [StorableObject] is the super class of all classes that'll be contained in the DB.
+ * It implements all the abstractions needed by the remote storage to do all the queries, i.e.
+ * - The [dbPath] that precedes every key for a given class (e.g. "objectX/"),
+ * - The converter (to local) that converts a T DB object to a local StorableObject<T>,
+ * - The converter (to DB)]that converts an instance of a local StorableObject<T> to a DB T.
+ *
+ * It is the role of every subclass to define the path and the conversion methods.
+ *
+ * This class takes a generic type T as argument, corresponding to the data class that is
```

Comment:
this is intended to be `class X : StorableObject<X>`, right?

Reply:
ah, no, from the conversions below it's not... so what kind of class is this expected to be used with? should `StorableObject` be an interface instead?

Reply:
As mentioned in the PR, you can check an example in `StorableObjectTest`, the `T` class in the one that is going to be stored in the DB and the `StorableObject<T>` the local one.

The reason I choose an abstract class was the use of our companion object :)